include/students.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+// --------------------------------------------------------------------------
+//This function accepts allStudent array as reference and adds to the array 
+//the remaining "credits" the student have 
+// --------------------------------------------------------------------------
+
+function addRemainingCredits(&$allStudents){                                          
+      
+    $creditsAmount = getCreditsAmount();                                              //Gets the amount of credit the students have from the credits table
+    $classesAmount = getClassesAmount();                                              //Same with classes amount
+
+    $studentIdsArray = array_column($classesAmount, 'StudentId');                   //Gets an array of studentsId from the classes array because is the easiest way I found to search for specific studentId.
+  
+    $i = 0;
+    foreach($creditsAmount as $credit){                                              //Go throught the creditsAmount array and
```

Comment:
A general thought/note: right now, you have nested `foreach` loops, meaning you have one foreach loop inside another foreach loop. When you do that, you introduce the potential for a pretty big performance hit in the future. Often, you won't notice it at first when you're writing the code, and it won't show up until you start to have a lot of data entered. This is because the performance here is exponential, instead of linear.

If you're familiar with Big O notation, this would be `O(n^2)` instead of `O(2n)`; if you don't know what that means, here's an attempt at summarizing this (feel free to skip this section if you know what I'm talking about):

> What that means is for every new item added to _each_ array, the amount of time it takes to run this code increases exponentially. So if you have 5 items in each array and the code takes 2 seconds to run, adding one more item will increase the execution time to 4 seconds. That's not so bad, right? Adding one more additional item, however, will increase the execution time to 16 seconds. That's not great, especially since it will escalate quickly from there.
> 
> On the other hand, linear growth would mean that each additional item only adds a constant amount of time (say 0.5 seconds). Using the examples above, the time to execute would increase to 2.5 seconds and 3 seconds, respectively. That's a pretty big difference in time, especially considering how fast computers are.
> 
> Now, the numbers I used above are just randomly picked for making a point -- the actual time here is probably significantly less, and won't start to really accelerate in growth until you reach a certain number of elements in the arrays. I wanted to try to illustrate the way that this could spiral into a big performance problem eventually, even if it's not currently an issue.

You can reduce the time complexity here by removing the nested loops and just having two `foreach` loops run consecutively. There's more than one way to do that, but the easiest (and best) way to do that in my opinion, is to create arrays that are indexed by a primary key. So in this case, you'd loop over the credits array and create a new array that has each student's credits already summed/grouped, and use the `studentId` as the key. Then you can loop over the students and access their information via key (and without an extra loop).

All that to say: this is totally fine for now. If you want to take a crack at performance stuff, this would be a great place to do it. I think being able to spot and fix obvious performance issues is a great skill to have as a developer, so I'd encourage you to at least think about how this could be restructured at some point in the summer.

Reply:
Yes! It makes total sense! I learned about time complexity in Data Structures last semester and I loved it! The teacher kept telling us how important Big O was and I'm glad to see that it is indeed. I'll work on this now!

Reply:
@robertlandis I started working on this in PR 2 but I'm realizing that I changed some things in PR 3, so these functions don't look the same although I still have the nested loops. I think is better if I do these changes directly in PR 3 if that's ok.